code/modules/mod/mod_activation.dm

```diff
@@ -167,6 +162,8 @@
 	mod_link.end_call()
 	to_chat(wearer, span_notice("MODsuit [active ? "shutting down" : "starting up"]."))
 	for(var/obj/item/part as anything in get_parts())
+		if(part.loc == src)
+			continue
 		var/datum/mod_part/part_datum = get_part_datum(part)
 		if(do_after(wearer, activation_step_time, wearer, MOD_ACTIVATION_STEP_FLAGS, extra_checks = CALLBACK(src, PROC_REF(get_wearer)), hidden = TRUE))
```

Comment:
this should be its own proc now and you should run it whenever someone tries to extend/retract a part of the turned on suit. also when turning on the suit the control unit should turn on first now (though when turning off the suit the control unit should still be last)